### PR TITLE
Update verifyrules to use a button and modal

### PR DESCRIPTION
**Pull request by celery6**
Lots of copy pasting, hope it's not too messy.

### CODE REVIEW

File: src/events/modalSubmitInteractionCreateEventHandler.ts
Comment (sunny-zuo) [inline src/events/modalSubmitInteractionCreateEventHandler.ts:25]: This seems like something that should be moved to the constructor, as we definitely don't need to re-build this every time we receive a modal submit interaction.

File: src/events/modalSubmitInteractionCreateEventHandler.ts
Comment (sunny-zuo) [inline src/events/modalSubmitInteractionCreateEventHandler.ts:45]: Would prefer logging before running the actual command. This way we'll still have logs if the execution doesn't finish successfully.

**Review by sunny-zuo — COMMENTED** [review]
Great start!

File: src/commands/chat/verification/verifyrules.ts
Comment (sunny-zuo) [inline src/commands/chat/verification/verifyrules.ts:28]:
```suggestion
            config.enableVerification ? 'enabled' : `disabled. Enable it using ${inlineCode('/config')}`
```

File: src/interactions/button/verifyRulesButton.ts
Comment (sunny-zuo) [inline src/interactions/button/verifyRulesButton.ts:13]: We can set a shorter cooldown for this, since showing a modal isn't very expensive and users could accidentally exit the modal. Perhaps something like 3 in 5 seconds.

File: src/commands/chat/verification/verifyrules.ts
Comment (sunny-zuo) [inline src/commands/chat/verification/verifyrules.ts:24]:
```suggestion
            new MessageButton().setCustomId(`verifyRules`).setLabel('Update Rules').setStyle('PRIMARY')
```

for consistency with our other buttons

**Review by sunny-zuo — APPROVED** [review]
Looks great, thanks!

**Comment by celery6** [comment]
Will merge once approved!

**Review by sunny-zuo — APPROVED** [review]
lgtm 🚀

**Pull request merged**
